sdk/core/core-client-rest/src/common.ts

```diff
@@ -94,6 +94,13 @@ export type RequestParameters = {
  */
 export type RawResponseCallback = (rawResponse: FullOperationResponse, error?: unknown) => void;
 
+/**
+ * Types of bodies supported on the response.
+ * NodeJS.ReadableStream and () =\> NodeJS.ReadableStream is Node only.
+ * Blob, ReadableStream<Uint8Array>, and () =\> ReadableStream<Uint8Array> are browser only.
+ */
+export type ResponseBodyType = RequestBodyType | Record<string, any>;
```

Comment:
does this have the effect of widening the response body to `Record<string, any>`? Is there a different way we might be able to model this?

Reply:
Ah great point. How about `Record<string, unknown>`? Looks like it is resistant to ["too much"](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgBrIN7IB4C5kgCuAtgEbTIC+AUGAJ4AOKAKsgLxrIA+yAShAgD2UACYAeAM5gooAOYAaZIRABrEIIDuIAHzVqQkFOQQAjPlYcseZCarIA9PbT7BhsMYBM59pmR18tpQOTvxCopLScorKapo6Lm7GAMzeHCYA3AlGEAAsqcgAFACU7NqYlEA) widening. Also, the customer will have to cast to a specific type anyway if they really need to access that property. In retrospect, providing the stream and optionally the string if that makes sense probably would have been sufficient. Parsing the body without a sensible associated type doesn't seem like the best customer experience.